lib/sneakers_handlers/exponential_backoff_handler.rb

```diff
@@ -70,53 +89,58 @@ def death_count(headers)
     end
 
     def log(message)
-      Sneakers.logger.debug do
-        "DelayedRetryHandler handler [queue=#{@primary_queue_name}] #{message}"
+      Sneakers.logger.info do
+        "[#{self.class}] #{message}"
       end
     end
 
-    def durable_exchanges?
-      options[:exchange_options][:durable]
-    end
-
-    def durable_queues?
-      options[:queue_options][:durable]
-    end
-
     def create_exchange(name)
       log("creating exchange=#{name}")
 
-      @channel.exchange(name, type: "topic", durable: durable_exchanges?)
-    end
-
-    def retry_exchange
-      @retry_exchange ||= create_exchange("#{options[:exchange]}.retry")
+      channel.exchange(name, type: "topic", durable: options[:exchange_options][:durable])
     end
 
     def primary_exchange
       @primary_exchange ||= create_exchange("#{options[:exchange]}")
     end
 
     def create_error_exchange!
-      @error_exchange ||= create_exchange("#{options[:exchange]}.error").tap do |exchange|
-        queue = @channel.queue("#{@queue.name}.error", durable: durable_queues?)
+      arguments = options[:queue_options][:arguments]
+
+      dlx_exchange_name = arguments.fetch("x-dead-letter-exchange")
+      dlx_routing_key = arguments.fetch("x-dead-letter-routing-key")
 
-        Array(@options[:routing_key]).each do |key|
-          queue.bind(exchange, routing_key: @queue.name + "." + key)
-          queue.bind(exchange, routing_key: @queue.name + "." + key + ".*")
-        end
+      @error_exchange ||= create_exchange(dlx_exchange_name).tap do |exchange|
+        queue = channel.queue("#{@queue.name}.error", durable: options[:queue_options][:durable])
+        queue.bind(exchange, routing_key: dlx_routing_key)
       end
     end
 
     def create_retry_queue!(delay)
-      @channel.queue("#{queue.name}.retry.#{delay}",
-        durable: durable_queues?,
-        arguments: {
-          :"x-dead-letter-exchange" => options[:exchange],
-          :"x-message-ttl" => delay * 1_000,
-          :"x-expires" => delay * 1_000 * 2
-        }
-      )
+      clear_queues_cache
+      channel.queue( "#{queue.name}.retry.#{delay}",
+         durable: options[:queue_options][:durable],
+         arguments: {
+           :"x-dead-letter-exchange" => options[:exchange],
+           :"x-dead-letter-routing-key" => queue.name,
+           :"x-message-ttl" => delay * 1_000,
+           :"x-expires" => delay * 1_000 * 2
```

Comment:
Wouldn't it be better to set x-expires to a fixed addition to the message-ttl, like 1 minute? I assume there is not much reason to keep it around after it has served it's purpose?

Also just to confirm, if new messages arrive say half way into the queue's expiry time, the queue is re-declared so the unused time starts counting again from 0?